### DISCUSSION TITLE
logs "Outcome: Success" & "Response succeeded." when returning an error

### DISCUSSION THREAD

**Discussion by tdelabro** [rwf2/Rocket#2310] [Questions]
Here are my logs:
```
 Aug 05 09:04:03.068 INFO DELETE /projects:
 Aug 05 09:04:03.068 INFO Matched: (delete_project) DELETE /projects
 Aug 05 09:04:03.068 INFO Outcome: Success
 Aug 05 09:04:03.068 INFO Response succeeded.
 ```
 There are all green (visually).
 
 And here is my code:
 ```
 #[delete("/projects")]
pub async fn delete_project() -> Result<(), ()> {
	Err(())
}
```

I return an error. I would expect that since my route returned an error, the logs would be red and not announcing to me the success of the call. Because the outcome of my call is not a success and the call literally failed. 

Is there something I don't understand there? Is this the expected behavior?

**Comment by the10thWiz** [comment]
The outcome & response reported are letting you know that the response was successfully sent to the client. You need to check on the client side to see whether a successful response was sent (and what the HTTP code was). Rocket doesn't log this by default, although you can create & attach a fairing to report this information.

`Outcome: Failure` means the request is then forwarded to an error catcher.

`Outcome: Foward` means the request is then forwarded to the next matching route.

`Response failed` means Rocket was unable to write the response to a client.
